libdnf/base/transaction.cpp

```diff
@@ -288,12 +288,14 @@ void Transaction::Impl::set_transaction(rpm::solv::GoalPrivate & solved_goal, Go
 
     // Add groups to the transaction
     for (auto & [group, action, reason] : solved_goal.list_groups()) {
-        groups.emplace_back(group, action, reason);
+        TransactionGroup tsgrp(group, action, reason);
```

Comment:
Because this ctor is not public anymore and it is a friend only in the `Transaction::Impl` scope.